### DISCUSSION TITLE
Single components import

### DISCUSSION THREAD

**Discussion by vitalijalbu** [ant-design/ant-design#42583] [General]
Hi, how I can import components one by one to reduce time loading ?
something like:

```
import Table from "antd/lib/table";
import Button from "antd/lib/button";
```

**Comment by RedJue** [comment]
What version do you use?

**Reply by vitalijalbu** [reply]
latest

**Reply by RedJue** [reply]
The latest version of antd will automatically tree-shaking, which will not affect the package size.

**Comment by BoyYangzai** [comment]
@vitalijalbu  
`import { Table } from "antd";` is enough 
because treeshaking is supported in lastest antd `es` production , don't need developer exact import by ourselves

If you want to know the `exact size` of the product of the access `antd` and whether the `treeshaking` was successful or not
you can use https://github.com/danvk/source-map-explorer

**Comment by zombieJ** [comment]
As @BoyYangzai reply. Let webpack to handle this. You do not need to care about it.
